### PR TITLE
Precompiled contracts for retrieving Theta balance and stake

### PR DESCRIPTION
**Pull request by jieyilong**

### CODE REVIEW

File: ledger/vm/contracts.go [hunk 1]
```diff
@@ -358,3 +361,37 @@ func (c *bn256Pairing) Run(input []byte) ([]byte, error) {
 	}
 	return false32Byte, nil
 }
+
+// thetaBalance retrieves the ThetaWei balance of the given address
+type thetaBalance struct {
+	evm *EVM
```
Comment (weiribao) [inline ledger/vm/contracts.go:367]: This field is not in use?

File: ledger/vm/contracts.go [hunk 2]
```diff
@@ -358,3 +361,37 @@ func (c *bn256Pairing) Run(input []byte) ([]byte, error) {
...
+func (c *thetaBalance) Run(evm *EVM, input []byte) ([]byte, error) {
+	address := common.BytesToAddress(input)
+	thetaBalance := evm.StateDB.GetThetaBalance(address)
+	thetaBalanceBytes := thetaBalance.Bytes()
+	return thetaBalanceBytes, nil
+}
+
+// thetaStake retrieves the total amount of ThetaWei the address staked to validators and/or guardians
+type thetaStake struct {
+	evm *EVM
```
Comment (weiribao) [inline ledger/vm/contracts.go:384]: Same as above.

File: ledger/vm/contracts.go (hunk 1 above)
Comment (jieyilong) [reply ledger/vm/contracts.go:367]: fixed

File: ledger/vm/contracts.go (hunk 2 above)
Comment (jieyilong) [reply ledger/vm/contracts.go:384]: fixed

**Pull request merged**